### DISCUSSION TITLE
blank visualization

### DISCUSSION THREAD

**Discussion by violacimatti** [gosling-lang/gosling.js#1046] [Q&A]
Hi, I have a .bw file of which I would like to visualize the mark bar, however, following also the tutorials, I always get a blank graph. 
The following is the code: 
`import gosling as gos 

data = gos.bigwig(
    url='/Users/violacimatti/Desktop/gosling/MB110_H3K27me3.bw',  
    column='position',
    value='peak',
    binSize=8
)

track = gos.Track(data).mark_bar().encode(
    x='position:G',
    y='peak:Q',
    size=gos.value(20),
    color=gos.value('blue')
).properties(width=725, height=180, layout='linear')

vis = track.view(title='group 4')
vis.save('group4.html')

`

**Comment by sehilyi** [comment]
Thank you for reaching out! The code looks good to me, and if I replace the URL with the one we have on gosling.js.org, it shows the visualization:

![Screenshot 2024-03-07 at 8 55 54 AM](https://github.com/gosling-lang/gosling.js/assets/9922882/bec76833-0e95-4056-8d59-a877435df5f1)


(Note that I replaced the last line (`vis.save('group4.html')`) with `vis` since `.save` exports the HTML file without rendering anything)

- You can find the code on this Colab Link: https://colab.research.google.com/gist/sehilyi/36b2b20b39d46bd8ddc94b4f82454b3b/multiple-coordinated-views.ipynb

Have you tried to open the `group4.html` file that you created as well?

**Reply by violacimatti** [reply]
Yes but as soon as I insert a local file path instead of an actual url it just gives me a blank graph

**Reply by sehilyi** [reply]
Hmm, I see. I wonder if data cannot be accessed by Gos. Could you try directly accessing the data that is hosted on the local server? When you print the `data`, you should be able to see the URL to your local data, something like

![Screenshot 2024-03-08 at 10 41 16 AM](https://github.com/gosling-lang/gosling.js/assets/9922882/989f5ccd-b4d5-4865-b805-d53df13d7aa9)

```
{'type': 'csv',
 'url': 'http://localhost:49967/resources/2529389-UCSC.HG38.Human.CytoBandIdeogram.bed',
 'headerNames': ['chrom', 'chromStart', 'chromEnd', 'name', 'stain'],
 'chromosomeField': 'chrom',
 'genomicFields': ['chromStart', 'chromEnd'],
 'separator': '\t'}
```

Opening the URL (e.g., `http://localhost:49967/resources/2529389-UCSC.HG38.Human.CytoBandIdeogram.bed` in my case) using your browser should download the file if it is accessible.

**Comment by violacimatti** [comment]
Yes, this works, but if try this code: 

```
import` gosling as gos 
import clodius as clodius 

data = gos.bigwig(
    url='http://localhost:8080/MB113_H3K27me3.BigWig',  
    column='position',
    value='peak',
    binSize=8
)

track = gos.Track(data).mark_area().encode(
    x='position:G',
    y='peak:Q',
    size=gos.value(2),
    color=gos.value('royalblue')
).properties(width=725, height=180, layout='linear')

vis = track.view(title='WNT')
vis.save('WNT.html')`
```

And then open the html file from the local server, this is what I get: 
<img width="731" alt="Screen Shot 2024-03-13 at 9 57 01 AM" src="https://github.com/gosling-lang/gosling.js/assets/127793264/c87f1f6d-e873-4a88-a729-b48ad81b4e70">
